### PR TITLE
Add AWS Foundational Security Best Practices RDS benchmark. Closes #116

### LINKED ISSUE #116: Add AWS Foundational Security Best Practices RDS benchmark
**Is your feature request related to a problem? Please describe.**
I'd like to be able to run the RDS benchmark for the AWS Foundational Security Best Practices framework.

**Describe the solution you'd like**
Implement RDS benchmark.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [x] Issue(s) linked

### CODE REVIEW

File: query/rds/rds_snapshot_private_access.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
+(
```
Comment (cbruno10) [inline query/rds/rds_snapshot_private_access.sql:1]: What does "private access" mean here? What is this query really checking for?

File: query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
+(
```
Comment (cbruno10) [inline query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:1]: `query/rds/rds_db_instance_and_cluster_enhanced_monitoring_enabled.sql`

File: query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+select
+  -- Required Columns
+  arn as resource,
+  case
+    when enabled_cloudwatch_logs_exports is not null then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when enabled_cloudwatch_logs_exports is not null then title || ' db cluster enhanced monitoring configured.'
```
Comment (cbruno10) [inline query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:10]:
```suggestion
    when enabled_cloudwatch_logs_exports is not null then title || ' enhanced monitoring enabled.'
```

File: query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    else title || ' db cluster enhanced monitoring not configured.'
```
Comment (cbruno10) [inline query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:11]:
```suggestion
    else title || ' enhanced monitoring not enabled.'
```

File: query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  -- Required Columns
+  arn as resource,
+  case
+    when class = 'db.m1.small' then 'skip'
+    when enhanced_monitoring_resource_arn is not null then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when class = 'db.m1.small' then title || ' enhanced monitoring not supported.'
+    when enhanced_monitoring_resource_arn is not null then title || ' instance enhanced monitoring configured.'
```
Comment (cbruno10) [inline query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:31]:
```suggestion
    when enhanced_monitoring_resource_arn is not null then title || ' enhanced monitoring configured.'
```

Comment (cbruno10) [reply query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:31]: We don't need the resource type in the reasons

File: query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    else title || ' instance enhanced monitoring not configured.'
```
Comment (cbruno10) [inline query/rds/rds_instance_and_cluster_configured_with_enhanced_monitoring.sql:32]:
```suggestion
    else title || ' enhanced monitoring not configured.'
```

File: query/rds/rds_db_instance_configured_with_iam_authentication.sql [hunk 7]
```diff
@@ -0,0 +1,16 @@
+select
```
Comment (cbruno10) [inline query/rds/rds_db_instance_configured_with_iam_authentication.sql:1]: `query/rds/rds_db_instance_iam_authentication_enabled.sql`

File: query/rds/rds_db_instance_configured_with_iam_authentication.sql [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  -- Required Columns
+  arn as resource,
+  case
+    when iam_database_authentication_enabled then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when iam_database_authentication_enabled then title || ' IAM authentication configured.'
```
Comment (cbruno10) [inline query/rds/rds_db_instance_configured_with_iam_authentication.sql:9]:
```suggestion
    when iam_database_authentication_enabled then title || ' IAM authentication enabled.'
```

File: query/rds/rds_db_instance_configured_with_iam_authentication.sql [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    else title || ' IAM authentication not configured.'
```
Comment (cbruno10) [inline query/rds/rds_db_instance_configured_with_iam_authentication.sql:10]:
```suggestion
    else title || ' IAM authentication not enabled.'
```

File: query/rds/rds_cluster_deletion_protection_enabled.sql [hunk 10]
```diff
@@ -0,0 +1,16 @@
+ select
```
Comment (cbruno10) [inline query/rds/rds_cluster_deletion_protection_enabled.sql:1]: `query/rds/rds_db_cluster_deletion_protection_enabled.sql`

We should be consistent with file names, most include `db`, e.g., `rds_db_cluster`, so this one should too

File: query/rds/rds_cluster_configured_with_iam_authentication.sql [hunk 11]
```diff
@@ -0,0 +1,16 @@
+select
```
Comment (cbruno10) [inline query/rds/rds_cluster_configured_with_iam_authentication.sql:1]: `query/rds/rds_db_cluster_configured_with_iam_authentication.sql`

File: query/rds/rds_cluster_configured_with_iam_authentication.sql [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  -- Required Columns
+  arn as resource,
+  case
+    when iam_database_authentication_enabled then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when iam_database_authentication_enabled then title || ' IAM authentication configured.'
```
Comment (cbruno10) [inline query/rds/rds_cluster_configured_with_iam_authentication.sql:9]:
```suggestion
    when iam_database_authentication_enabled then title || ' IAM authentication enabled.'
```

File: query/rds/rds_cluster_configured_with_iam_authentication.sql [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    else title || ' IAM authentication not configured.'
```
Comment (cbruno10) [inline query/rds/rds_cluster_configured_with_iam_authentication.sql:10]:
```suggestion
    else title || ' IAM authentication not enabled.'
```

File: query/rds/rds_cluster_and_database_snapshot_encrypted_at_rest.sql [hunk 14]
```diff
@@ -0,0 +1,37 @@
+(
```
Comment (cbruno10) [inline query/rds/rds_cluster_and_database_snapshot_encrypted_at_rest.sql:1]: `query/rds/rds_db_snapshot_encrypted_at_rest.sql`

File: query/rds/rds_cluster_and_database_snapshot_encrypted_at_rest.sql [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+select
+  -- Required Columns
+  arn as resource,
+  case
+    when storage_encrypted then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when storage_encrypted then title || ' db cluster snapshot encrypted.'
```
Comment (cbruno10) [inline query/rds/rds_cluster_and_database_snapshot_encrypted_at_rest.sql:10]:
```suggestion
    when storage_encrypted then title || ' encrypted at rest.'
```

Please make similar changes for the other reasons below

File: query/rds/rds_automatic_minor_version_upgrade_enabled.sql [hunk 16]
```diff
@@ -0,0 +1,16 @@
+select
```
Comment (cbruno10) [inline query/rds/rds_automatic_minor_version_upgrade_enabled.sql:1]: `query/rds/rds_db_instance_automatic_minor_version_upgrade_enabled.sql`

File: query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql [hunk 17]
```diff
@@ -0,0 +1,17 @@
+select
```
Comment (cbruno10) [inline query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql:1]: `query/rds/rds_db_cluster_aurora_backtracking_enabled.sql`

File: query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+  -- Required Columns
+  arn as resource,
+  case
+    when engine not ilike '%aurora-mysql%' then 'skip'
+    when backtrack_window is not null then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when engine not ilike '%aurora-mysql%' then title || ' not Aurora MySQL-compatible edition.'
+    when backtrack_window is not null then title || ' Aurora MySQL-compatible cluster configured with backtracking.'
```
Comment (cbruno10) [inline query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql:11]:
```suggestion
    when backtrack_window is not null then title || ' backtracking enabled.'
```

File: query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+    else title || ' Aurora MySQL-compatible cluster not configured with backtracking.'
```
Comment (cbruno10) [inline query/rds/rds_amazon_aurora_cluster_backtracking_enabled.sql:12]:
```suggestion
    else title || '  backtracking not enabled.'
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comments

File: query/rds/rds_snapshot_private_access.sql (hunk 1 above)
Comment (khushboo9024) [reply query/rds/rds_snapshot_private_access.sql:1]: @cbruno10 `private access` here generally  means that the snapshots should not be publicly accessible for restoration etc. We already have this query `rds_snapshot_prohibit_public_access.sql`(which is same as `rds_snapshot_private_access.sql` query) but for the reason change we have added a new query file as the title in FSBP is `[RDS.1] RDS snapshots should be private`.
https://turbothq.slack.com/archives/C01RN2G6KNK/p1622200538038400

Comment (cbruno10) [reply query/rds/rds_snapshot_private_access.sql:1]: We should just use `rds_snapshot_prohibit_public_access.sql` for this, as the control titles per framework will be different anyway, so we should rely on good base query naming/reuse.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**